CMakeLists.txt

```diff
@@ -280,8 +280,8 @@ if (USE_GNUTLS)
 	endif()
 	target_compile_definitions(datachannel PRIVATE USE_GNUTLS=1)
 	target_compile_definitions(datachannel-static PRIVATE USE_GNUTLS=1)
-	target_link_libraries(datachannel PRIVATE GnuTLS::GnuTLS)
-	target_link_libraries(datachannel-static PRIVATE GnuTLS::GnuTLS)
+	target_link_libraries(datachannel PUBLIC GnuTLS::GnuTLS)
+	target_link_libraries(datachannel-static PUBLIC GnuTLS::GnuTLS)
```

Comment:
Why is that change necessary? GnuTLS, OpenSSL and libnice dependencies should not be linked against but not exposed to dependencies for linking.